validation/mock_metric_validation.go

```diff
@@ -0,0 +1,56 @@
+package validation
+
+import (
+	"fmt"
+	"reflect"
+
+	"google.golang.org/genproto/googleapis/monitoring/v3"
+	"google.golang.org/genproto/googleapis/rpc/errdetails"
+)
+
+const (
+	missingFieldMsg = "%v must contain requried %v field"
+)
+
+func IsValidRequest(req interface{}) error {
+	reqReflect := reflect.ValueOf(req)
+	requiredFields := []string{"Name"}
+	br := &errdetails.BadRequest{}
+	requestName := ""
+
+	switch req.(type) {
+	case *monitoring.CreateMetricDescriptorRequest:
+		requiredFields = append(requiredFields, "MetricDescriptor")
+		requestName = "CreateMetricDescriptor"
+	case *monitoring.GetMetricDescriptorRequest:
+		requestName = "GetMetricDescriptorRequest"
+	case *monitoring.DeleteMetricDescriptorRequest:
+		requestName = "DeleteMetricDescriptorRequest"
+	case *monitoring.GetMonitoredResourceDescriptorRequest:
+		requestName = "GetMonitoredResourceDescriptorRequest"
+	case *monitoring.ListMetricDescriptorsRequest:
+		requestName = "ListMetricDescriptorsRequest"
+	}
+
+	for _, field := range requiredFields {
```

Comment:
Since this part is exactly the same (I think) as the validation for trace missing fields, can we reuse it? Maybe turn it into a function that takes in `requiredFields` and `reqReflect` as parameters and runs the check